### DISCUSSION TITLE
Distinguish merge commits when checking commit message.

### DISCUSSION THREAD

**Discussion by donquixote** [phpro/grumphp#897] [Ideas]
| Q               | A
| --------------- | ---
| Version         | GrumPHP 0.15.2
| Bug?            | no
| New feature?    | yes
| Question?       | no
| Documentation?  | no
| Related tickets | 

[class CommitMessage](https://github.com/phpro/grumphp/blob/f732427969907c25a27e0f6cdd1f8da6233ea7d9/src/Task/Git/CommitMessage.php#L78) analyses if the commit message matches a given pattern. The exact rules can be configured.

Typically, a merge commit will have a different commit message than a regular commit.
E.g. `Merge branch 'a' into b.` vs `Issue #55: Add more tests.`.

Currently grumphp does not distinguish between actual merge commits and regular commits.
This means it is possible to (accidentally) name a non-merge commit like `Merge 'a' into b.` without grumphp complaining.

It would be great if grumphp could distinguish between these cases.

If we want to be excessive we could also distinguish multi-parent commits :) but let's not overdo it!

**Comment by veewee** [comment]
Currently the merge patterns are ok. I agree that it allows you to bypass the system, but so does adding the `-n` option to a `git commit`.

In order to detect if current commit is an acutal merge commit, additional git commands need to be executed. (Checking if the commit contains multiple parents).

Feel free to provide a solution for this.
